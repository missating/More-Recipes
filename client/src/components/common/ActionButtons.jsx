import React from 'react';
import PropTypes from 'prop-types';


const ActionButtons = props => (
  <div className="container">

    <button className="btn disabled">
      <span>
        <i className="far fa-thumbs-up" />
      </span>
      {props.singleRecipe.downvote}
    </button>

    <button className="btn disabled">
      <span>
        <i className="fa fa-thumbs-down" />
      </span>
      {props.singleRecipe.upvote}
    </button>

    <button className="btn disabled">
      <span>
        <i className="far fa-heart" />
      </span>
      {props.singleRecipe.favourite}
    </button>
  </div>
);

ActionButtons.propTypes = {
  singleRecipe: PropTypes.shape({
    favourite: PropTypes.number,
    upvote: PropTypes.number,
    downvote: PropTypes.number
  }).isRequired
};

export default ActionButtons;

