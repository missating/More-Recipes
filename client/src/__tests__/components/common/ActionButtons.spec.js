import React from 'react';
import toJSON from 'enzyme-to-json';
import ActionButtons from '../../../components/common/ActionButtons';

describe('Action Buttons component', () => {
  const props = {
    singleRecipe: {
      favourite: 1,
      upvote: 1,
      downvote: 1
    }
  };

  it('Should render', () => {
    const wrapper = shallow(<ActionButtons
      {...props}
    />);
    expect(wrapper.find('div').length).toEqual(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

