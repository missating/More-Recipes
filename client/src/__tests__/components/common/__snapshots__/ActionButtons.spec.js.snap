// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Buttons component Should render 1`] = `
<div
  className="container"
>
  <button
    className="btn disabled"
  >
    <span>
      <i
        className="far fa-thumbs-up"
      />
    </span>
    1
  </button>
  <button
    className="btn disabled"
  >
    <span>
      <i
        className="fa fa-thumbs-down"
      />
    </span>
    1
  </button>
  <button
    className="btn disabled"
  >
    <span>
      <i
        className="far fa-heart"
      />
    </span>
    1
  </button>
</div>
`;
